crates/engine/tree/src/backfill.rs

```diff
@@ -10,27 +10,27 @@
 use reth_stages_api::{ControlFlow, PipelineError, PipelineTarget};
 use std::task::{Context, Poll};
 
-/// A handler for the pipeline.
-pub trait PipelineHandler: Send + Sync {
-    /// Performs an action on the pipeline.
-    fn on_action(&mut self, event: PipelineAction);
+/// Backfill sync mode functionality.
+pub trait BackfillSync: Send + Sync {
+    /// Performs a backfill action.
+    fn on_action(&mut self, event: BackfillAction);
 
     /// Polls the pipeline for completion.
-    fn poll(&mut self, cx: &mut Context<'_>) -> Poll<PipelineEvent>;
+    fn poll(&mut self, cx: &mut Context<'_>) -> Poll<BackfillEvent>;
 }
 
-/// The actions that can be performed on the pipeline.
+/// The backfill actions that can be performed.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub enum PipelineAction {
-    /// Start the pipeline with the given target.
+pub enum BackfillAction {
+    /// Start backfilling with the given target.
     Start(PipelineTarget),
```

Comment:
this type we can keep as `PipelineTarget so we don't need to change existing code.